src/components/Navigation/components/Item/Item.tsx

```diff
@@ -38,6 +38,7 @@ interface SecondaryAction {
 
 export interface Props extends ItemURLDetails {
   icon?: IconProps['source'];
```

Comment:
Does the type of `icon` have to be updated to accept a string?

Reply:
`IconProps['source']` is the type for the source prop of  `<Icon> which already accepts a string from your work on untrusted icons.